packages/devtools_app/lib/src/extensions/extension_service.dart

```diff
@@ -268,22 +277,44 @@ class ExtensionService extends DisposableController
     // Set the enabled state for all matching extensions, even if some are
     // marked as ignored due to being a duplicate. This ensures that
     // devtools_options.yaml files are kept in sync across the project.
-    final allMatchingExtensions = [..._runtimeExtensions, ..._staticExtensions]
-        .where((e) => e.name == extension.name);
-    for (final ext in allMatchingExtensions) {
-      await server.extensionEnabledState(
-        devtoolsOptionsFileUri: ext.devtoolsOptionsUri,
-        extensionName: ext.name,
-        enable: enable,
-      );
-    }
+    final allMatchingExtensions = [
+      ...runtimeExtensions,
+      ...staticExtensions,
+    ].where((e) => e.name == extension.name);
+    await Future.wait([
+      for (final ext in allMatchingExtensions)
+        server.extensionEnabledState(
+          devtoolsOptionsFileUri: ext.devtoolsOptionsUri,
+          extensionName: ext.name,
+          enable: enable,
+        ),
+    ]);
     await _refreshExtensionEnabledStates();
   }
 
+  /// Marks this extension configuration as ignored or unignored based on the
+  /// value of [ignore].
+  ///
+  /// An extension may be ignored if it is a duplicate or if it is an older
+  /// version of an existing extension, for example.
+  void ignoreExtension(DevToolsExtensionConfig ext, [bool ignore = true]) {
```

Comment:
I think this function name is confusing. It's fine for the case where we want to mark `ext` as ignored, but it's not obvious that `ignoreExtension(fooExt, false)` is marking the extension as not ignored. Making the optional positional parameter a named parameter would help a bit, but maybe it's worth considering renaming this function to something like `setExtensionIgnored(DevToolsExtensionConfig ext, {required bool ignore})`.